zjit/src/codegen.rs

```diff
@@ -172,7 +186,16 @@ fn gen_entry(cb: &mut CodeBlock, iseq: IseqPtr, function: &Function, function_pt
     asm.frame_teardown();
     asm.cret(C_RET_OPND);
 
-    asm.compile(cb).map(|(start_ptr, _)| start_ptr)
+    let result = asm.compile(cb).map(|(start_ptr, _)| start_ptr);
+    if let Some(start_addr) = result {
+        if get_option!(perf) {
+            let start_ptr = start_addr.raw_ptr(cb) as usize;
+            let end_ptr = cb.get_write_ptr().raw_ptr(cb) as usize;
+            let code_size = end_ptr - start_ptr;
+            register_with_perf("entry".into(), start_ptr, code_size);
```

Comment:
I think it's helpful to include the ISEQ name in the symbol for the entry, also keeping "entry" as part of the symbol. We'd be interested in which ISEQs are called from C, and using the same "entry" symbol for all entries could be problematic depending on how you aggregate perf profiles.

Reply:
Will do before landing